src/api/import/import.middlewares.js

```diff
@@ -0,0 +1,97 @@
+/* eslint-disable no-await-in-loop */
+import axios from 'axios';
+import { parse } from 'csv-parse/sync';
+
+import catalog from '../commons/catalog';
+import { BadRequestError } from '../commons/http-errors';
+import controllers from '../commons/middlewares/crud.middlewares';
+import readQuery from '../commons/queries/structures.query';
+import { structuresRepository as repository } from '../commons/repositories';
+
+const CSV_DELIMITER = ';';
+const statusMapping = {
+  O: 'active',
+  F: 'inactive',
+  P: 'forthcoming',
+};
+
+const chunkArray = (array, chunkSize) => array.reduce((resultArray, item, index) => {
+  const chunkIndex = Math.floor(index / chunkSize);
+  if (!resultArray[chunkIndex]) {
+    // eslint-disable-next-line no-param-reassign
+    resultArray[chunkIndex] = [];
+  }
+  resultArray[chunkIndex].push(item);
+  return resultArray;
+}, []);
+
+// eslint-disable-next-line no-promise-executor-return
+const sleep = (ms) => new Promise((resolve) => setTimeout(resolve, ms));
+
+const parseOptions = {
+  delimiter: CSV_DELIMITER,
+  trim: true,
+  columns: true,
+};
+
+export async function bulkImportStructures(req, res, next) {
+  const data = parse(req.files[0].buffer, parseOptions);
+  const structures = [];
+  const structureIds = [];
+  for (let i = 0; i < data.length; i += 1) {
+    const item = data[i];
+    const structure = {
+      acronymEn: item['Nom court  en anglais :'],
+      acronymFr: item['Sigle en français'],
+      acronymLocal: item['Nom court officiel dans la langue originale'],
+      closureDate: item['Date de fermeture {2020-07-02}'],
+      creationDate: item['Date de création {2020-07-02}'],
+      nameEn: item['Nom long en anglais :'],
+      officialName: item['Nom officiel dans la langue originale'],
+      otherNames: item['Autres dénominations possibles séparées par des ;'].split(';'),
+      shortName: item['Nom court en français'],
+      structureStatus: statusMapping[item['Statut [O = Ouvert, F = Fermé, P = Potentiel]']],
+      usualName: item['Nom usuel en français'],
+    };
+    req.body = structure;
```

Comment:
@folland87 I would like to use the router to call the `/structures` endpoint here to create a new one. But no idea how to do that.

Reply:
Inside your controller, you could make a POST `${req.hostname}/structures`, i guess.
Not sure, however, that we should implement an multipart/form-data import with csv file on the server, or let the frontend do the job of transforming csv to json and implement an application/json endpoint.

Reply:
If we implement an API endpoint, it would be useful for external services. Let's discuss about that.